lib/cli/config.js

```diff
@@ -76,6 +77,10 @@ exports.loadConfig = filepath => {
     } else {
       config = parsers.json(filepath);
     }
+
+    config = yargs(undefined, path.dirname(filepath))
+      .parserConfiguration(require('./options').YARGS_PARSER_CONFIG)
+      .config(config);
```

Comment:
Noting https://github.com/mochajs/mocha/pull/4407/files#r1509449877 here: would be nice to have discussion on which API option to use. I'm leaning towards this one, but want to hear if there are advantages to `applyExtends` I missed.

Reply:
I'm not entirely sure anymore why I decided to use this function instead.
I remember that when using a different approach, some options were overridden by default.

I will read through my commit messages during the upcoming weekend. I really hope I can reproduce my thoughts this way.